aptos-move/aptos-vm/src/keyless_validation.rs

```diff
@@ -153,6 +174,9 @@ pub(crate) fn validate_authenticators(
         {
             return Err(VMStatus::error(StatusCode::FEATURE_UNDER_GATING, None));
         }
+        if matches!(pk, AnyKeylessPublicKey::Federated { .. }) && !features.is_federated_keyless_enabled() {
```

Comment:
now it seems this gating is not needed...

The new txn variant can't get agreed on and reach here (execution), unless the majority of validators have the new binary.

sorry for the back and forth...

Reply:
Oh, no, it is needed to enable the feature via governance!

Reply:
ok, i was only talking about the feature-gating for backward compatibility purpose.

If there are other reasons, let's keep it then.